cli/cli-client/src/main/java/org/infinispan/cli/patching/PatchTool.java

```diff
@@ -0,0 +1,470 @@
+                  throw new RuntimeException(e);
+               }
+            }
+         }
+      });
+      targetFiles.forEach((k2, targetFile) -> {
+         if (!v1Files.containsKey(k2)) {
+            operations.add(PatchOperation.add(targetFile.getVersionedPath(), targetFile.getDigest(), targetFile.getPermissions()));
+            addFileToZip(zipfs, target, targetFile);
+         }
+      });
+
+      // Write out the JSON patch file
+      Path patchPath = zipfs.getPath("patch-" + patchInfo.getSourceVersion() + "_" + patchInfo.getTargetVersion() + ".json");
+      try (OutputStream os = Files.newOutputStream(patchPath, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)) {
+         mapper.writerWithDefaultPrettyPrinter().writeValue(os, patchInfo);
+      }
+   }
+
+   private Map<Path, ServerFile> getServerFiles(Path base) throws IOException {
+      Pattern IGNORE = Pattern.compile("^(\\.patches/|server/data/|server/log/|server/lib/).*$");
+      Pattern SOFT_REPLACE_PATTERN = Pattern.compile("^server/conf/.*$");
+      Map<Path, ServerFile> files = new TreeMap<>();
+      Files.walkFileTree(base, new SimpleFileVisitor<Path>() {
+         @Override
+         public FileVisitResult visitFile(Path oPath, BasicFileAttributes attrs) {
+            Path rPath = base.relativize(oPath);
+            String rPathName = rPath.toString();
+            if (!IGNORE.matcher(rPathName).matches()) {
+               ServerFile file = new ServerFile(rPath, sha256(oPath), getPermissions(oPath), SOFT_REPLACE_PATTERN.matcher(rPathName).matches());
+               files.put(file.getUnversionedPath(), file);
+            }
+            return FileVisitResult.CONTINUE;
+         }
+      });
+      return files;
+   }
+
+   private static String getPermissions(Path path) {
+      try {
+         return PosixFilePermissions.toString(Files.getPosixFilePermissions(path));
+      } catch (IOException e) {
+         throw new RuntimeException(e);
+      }
+   }
+
+   private List<PatchInfo> getPatchInfos(FileSystem zipfs) throws IOException {
+      List<Path> paths = Files.find(zipfs.getPath("/"), 1,
+            (p, a) -> a.isRegularFile() && p.getFileName().toString().startsWith("patch-") && p.getFileName().toString().endsWith(".json")
+      ).collect(Collectors.toList());
+      List<PatchInfo> infos = new ArrayList<>(paths.size());
+      for (Path path : paths) {
+         try (InputStream is = Files.newInputStream(path, StandardOpenOption.READ)) {
+            infos.add(mapper.readValue(is, PatchInfo.class));
+         }
+      }
+      return infos;
+   }
+
+   private Properties getVersionProperties(Path base) throws IOException {
+      // Load the META-INF/infinispan-version.properties from the lib/infinispan-commons jar
+      Path lib = base.resolve("lib");
+      File[] commons = lib.toFile().listFiles((dir, name) -> name.startsWith("infinispan-commons-") && name.endsWith(".jar"));
+      if (commons.length != 1) {
```

Comment:
```suggestion
      if (commons == null || commons.length != 1) {
```
`commons` can be `null` if you set the wrong path it can't find the commons.